src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java

```diff
@@ -215,19 +216,12 @@ public int[] getRequiredTokens() {
         return new int[] {TokenTypes.PACKAGE_DEF, TokenTypes.IMPORT, TokenTypes.STATIC_IMPORT, };
     }
 
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
-    @SuppressWarnings("deprecation")
     @Override
     public void beginTree(DetailAST rootAST) {
+        final String fullFileName = getFilePath();
```

Comment:
please move close to first usage, as close as possible.

Reply:
done

Reply:
done